videos/videos.js

```diff
@@ -31,94 +31,111 @@ router.get("/:id", async (req, res) => {
+                } else {
+                  // Start with request to DS server, should get a 200 immediately if working
+                  // Then go to rest of functions
+                  // Potentially explore
+
+                  const song = await Songs.add(songObject);
+                  // console.log(song);
+
+                  const videoObjectComplete = { ...videoObject, song_id: song };
+
+                  // console.log(videoObjectComplete);
+
+                  // console.log(videoObjectComplete);
+                  // we want song to return its id
+                  // then we'll destructure or spread some object to add that in
+                  // then we'll 'add' that object to videos.add
+                  const video = await Videos.add(videoObjectComplete);
+                  console.log(video);
+                  axios
+                    .post(
+                      "http://sample.eba-5jeurmbw.us-east-1.elasticbeanstalk.com/entry",
+                      null,
+                      {
+                        params: {
+                          preview: preview,
+                          video_id: video,
+                        },
+                      }
+                    )
+                    .catch((err) => console.log(err));
+
+                  objectIds = {
+                    songId: song,
+                    videoId: video,
+                  };
+
+                  console.log(objectIds);
+
+                  // console.log(video);
+
+                  res.status(200).json(objectIds);
+                }
               }
             }
-          )
-          .catch((err) => console.log(err));
-
-        objectIds = {
-          songId: song,
-          videoId: video
-        };
-
-        console.log(objectIds);
-
-        // console.log(video);
-
-        res.status(200).json(objectIds);
-        // if (!video_title) {
-        //   res.status(400).json({ message: "PLease provide a Video Title!" });
-        // } else {
-        //   if (!location) {
-        //     res.status(400).json({ message: "Please provide a location!" });
-        //   } else {
-        //     if (!song_id) {
-        //       res.status(400).json({ message: "Please provide a song!" });
-        //     } else {
-        //       if (!user_id) {
-        //         res.status(400).json({ message: "Please provide a user!" });
-        //       } else {
-        //         const video = await Videos.add(data);
-        //         res.status(200).json(video);
-        //       }
-        //     }
-        //   }
-        // }
-      } catch (err) {
-        console.log(err);
-        res.status(500).json({ message: "Try again later!", err });
+          }
+        }
       }
+    }
+  } catch (err) {
+    console.log(err);
+    res.status(500).json({ message: "Try again later!", err });
+  }
 });
 
-router.put("/:id", (req, res) => {
+router.put("/:id", restricted, (req, res) => {
```

Comment:
We can ultimately prob kill this PUT, but let's keep it around in case we need it again